packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,116 @@
+    const cameledKey = _.camelCase(key);
+    newList.push({
+      cloudFormationParameterName: cameledKey,
+      environmaneVariableName: key,
+    });
+    newReferences[key] = { Ref: cameledKey };
+    newKeyValue[cameledKey] = value;
+  });
+
+  updateStoredList(resourceName, newList);
+  updateStoredReferences(resourceName, newReferences);
+  updateStoredKeyValue(context, resourceName, newKeyValue);
+};
+
+export const deleteEnvironmentVariable = (context: $TSContext, resourceName: string, targetedKey: string): void => {
+  let newList = getStoredList(resourceName);
+  const newReferences = getStoredReferences(resourceName);
+  const newKeyValue = getStoredKeyValue(context, resourceName);
+  const cameledKey = _.camelCase(targetedKey);
+  newList = _.filter(newList, item => {
+    return item.cloudFormationParameterName !== cameledKey && item.environmaneVariableName !== targetedKey;
+  });
+  _.unset(newReferences, targetedKey);
+  _.unset(newKeyValue, cameledKey);
+
+  updateStoredList(resourceName, newList);
+  updateStoredReferences(resourceName, newReferences);
+  updateStoredKeyValue(context, resourceName, newKeyValue);
+};
+
+const getStoredList = (resourceName: string): { cloudFormationParameterName: string; environmaneVariableName: string }[] => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
+  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as object) || {};
+  return _.get(functionParameters, 'environmentVariableList', []);
+};
+
+const updateStoredList = (resourceName: string, newList: object): void => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
+  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as object) || {};
+  _.set(functionParameters, 'environmentVariableList', newList);
+  JSONUtilities.writeJson(functionParameterFilePath, functionParameters);
+};
+
+const getStoredReferences = (resourceName: string): object => {
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
+  const cfnFileName = `${resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourcePath, cfnFileName);
+  const cfnContent = JSONUtilities.readJson(cfnFilePath, { throwIfNotExist: false }) || {};
+  return _.get(cfnContent, ['Resources', 'LambdaFunction', 'Properties', 'Environment', 'Variables'], {});
+};
+
+const updateStoredReferences = (resourceName: string, newReferences: object): void => {
```

Comment:
this function also needs to set the CFN parameter

Reply:
As I commented that injection envVars into `process.env` acctually worked without CFN parameters. Does prameters expect to have all keys of environment variables?

Reply:
it doesn't need the keys, it needs the values (which are in CFN parameters). When an env var is defined as:
```json
{
  "Resources": {
    "LambdaFunction": {
      "Properties": {
        "Environment": {
          "Variables": {
            "MY_ENV_VAR": {
              "Ref": "myEnvVar"
            }
          }
        }
      }
    }
  }
}
```
There must be a parameter in the template called "myEnvVar" that the ref is referencing:
```json
{
  "Parameters": {
    "myEnvVar": {
      "Type": "string"
    }
  }
}
```
Then when Amplify does the push, it has a mechanism for passing in the parameter value from `team-provider-info.json`